skdownscale/spatial_models/sd.py

```diff
@@ -0,0 +1,124 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+
+class SpatialDisaggregator:
+    """
+    Spatial disaggregation model class
+    Apply spatial disaggregation algorithm to an xarray Dataset with fit
+    and predict methods using NASA-NEX method for spatial disaggregation
+    (see Thrasher et al, 2012).
+
+    Parameters
+    ----------
+    var : str
+          specifies the variable being downscaled. Default is
+          temperature and other option is precipitation.
+    """
+
+    def __init__(self, var='temperature'):
+        self._var = var
+
+        if var == 'temperature':
+            pass
+        elif var == 'precipitation':
+            pass
+        else:
+            raise NotImplementedError(
+                'functionality for spatial disaggregation' ' of %s has not yet been added' % var
+            )
+
+    def fit(self, ds_bc, climo_coarse, var_name, lat_name='lat', lon_name='lon'):
+        """
+        Fit the scaling factor used in spatial disaggregation
+
+        Parameters
+        -----------
+        ds_bc : xarray.Dataset
+                Daily bias corrected data at the model resolution
+        climo_coarse : xarray.DataArray or xarray.Dataset
+                Observed climatology that has been regridded (coarsened)
+                to the model resolution
+        var_name : str
+                  Specifies the data variable name within ds_bc of the
+                  daily bias corrected data
+        lat_name : str
+                  Name of the latitude dimension of ds_bc and climo_coarse,
+                  default is 'lat'.
+        lon_name : str
+                  Name of the longitude dimension of ds_bc and climo_coarse,
+                  default is 'lon'.
+        """
+
+        # check that climo has been regridded to model res
+        if not np.array_equal(ds_bc[lat_name], climo_coarse[lat_name]):
+            raise ValueError('climo latitude dimension does not match model res')
+        if not np.array_equal(ds_bc[lon_name], climo_coarse[lon_name]):
+            raise ValueError('climo longitude dimension does not match model res')
```

Comment:
maybe something like this:
```suggestion
        if not ds_bc[lat_name].equals(climo_coarse[lat_name]):
            raise ValueError('climo latitude coordinate does not match model res')
        if not ds_bc[lon_name].equals(climo_coarse[lon_name]):
            raise ValueError('climo longitude coordinate does not match model res')
```